app/vmalert/remotewrite/remotewrite.go

```diff
@@ -10,4 +10,6 @@ type RWClient interface {
 	Push(s prompbmarshal.TimeSeries) error
 	// Close stops the client. Client can't be reused after Close call.
 	Close() error
+	// CloseAndReport stops the client and reports dropped rows
+	CloseAndReport() (int64, error)
```

Comment:
wdyt about modifying the Close implementation instead, so it could return an error if the last flush attempt has failed? If we do so, it would be pretty straightforward change.

Reply:
It's not only about last flush, in replay mode, it could be some write failures in the middle; 
and in non-replay mode, we don't care about the dropRows when stopping group, but need to catch possible bug
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/fb7454a14d1f5c014914ce3a192b99e62490d768/app/vmalert/remotewrite/client.go#L142-L144

Reply:
> It's not only about last flush, in replay mode, it could be some write failures in the middle;

This implementation doesn't catch rules in the middle, unless the RW client queue is full. But this error will be returned and before this patch.

>and in non-replay mode, we don't care about the dropRows when stopping group

Maybe we should care. Log a line that last attempt to push data didn't succeed is better than silently exit.